tests/integration/integration_test.go

```diff
@@ -314,50 +314,59 @@ func getTokenForLoggedInUser(t testing.TB, session *TestSession, scopes ...auth.
 	return token
 }
 
-func NewRequest(t testing.TB, method, urlStr string) *http.Request {
+func NewRequest(t testing.TB, method, urlStr string, token ...string) *http.Request {
```

Comment:
@go-gitea/maintainers 

I added an optional `token` parameter to keep the other test code cleaner. Not adding this parameter would require a call to `addTokenAuthHeader` after every `NewRequest...` call. The downside is we can't use it with the `NewRequestf` method because this method already takes variadic parameters. Therefore I would convert all `NewRequestf` calls to `NewRequest(..., fmt.Sprintf(...))`. I think that's ok because the other methods like `NewRequestWithBody` also don't have a `...f` version.

Converting all calls (200+) will take some time so I would like to get a decision before continuing.
For reference: There are 1300 non-`...f` calls and 450 urls with `?token=` parameter which would need the `addTokenAuthHeader` call.

1. Add `addTokenAuthHeader` when needed
2. Remove `NewRequestf` and use optional `token` parameter

Reply:
1 is more explicit/logical to me, but I think either option is fine

Reply:
The chained call:

```
NewRequest(...).WithToken(...)
```

Reply:
Thought about a builder pattern too but `NewRequest` returns a `http.Request` for which we can't add such a method. If we use a new return type we have to adjust all other places which use the created request before sending it.

Edit: Had a look now and the builder should be possible too. There is a manageable amount of places to adjust for a new type (adding headers to `req.Headers`, 'AddBasicAuthHeader' calls, ...).

Reply:
So builder it is?

Reply:
Perhaps. Converting the `NewRequestf` calls is just a regex replace but the builder may be more flexible in the long term which should take precedence.

Reply:
Use a wrap of request?

Reply:
Changed the code to use a builder pattern. Please have a look if I should continue with that.